[Mandrel 25.0.x future-defaults=all, Graal master] All quarkus native integration tests fail since 2025-12-19

### Describe the bug

The last clean build (when all quarkus native integration tests passed) was from December 18, 2025 here:
https://github.com/graalvm/mandrel/actions/runs/20321891451

This is using revision `e98bc519e1abd959bd1ffc8cf84b94100eadda0d` from quarkus.

Since December 19, 2025 or quarkus revision `614cd6f798646765ab0e1791528c820df4798cf7` all quarkus native integration tests fail:
https://github.com/graalvm/mandrel/actions/runs/20355898577

It doesn't seem GraalVM master revision related as I ran the same local GraalVM master build when verifying #51538. Now with a newer quarkus version it fails.

**Steps to reproduce:**

1. Grab a graal-master build from the passing CI run [here](https://github.com/graalvm/mandrel/actions/runs/20321891451/artifacts/4906257520)
2. Build quarkus main, revision `614cd6f798646765ab0e1791528c820df4798cf7` or newer.
3. Run any native integration test (e.g. `smallrye-config`)
4. Observe that the native binary doesn't start.

### Expected behavior

Quarkus native integration tests pass.

### Actual behavior

Quarkus native integration tests fail.

Running the executable manually reveals a bit more:

```
$ /home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/integration-tests/smallrye-config/target/quarkus-integration-test-smallrye-config-999-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 '-Dtest.url=http://localhost:${quarkus.http.test-port:8081}' -Dquarkus.log.file.path=/home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/integration-tests/smallrye-config/target/quarkus.log -Dquarkus.log.file.enabled=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.profile=prod -Dquarkus.native.enable-reports=true -Dquarkus.native.container-build=false
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2025-12-19 19:04:31,674 WARN  [io.quarkus.config] (main) Unrecognized configuration key "quarkus.log.category.io.quarkus.level" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2025-12-19 19:04:31,680 ERROR [io.quarkus.runtime.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:105)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:78)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:142)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.RuntimeException: Unable to start HTTP server
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:984)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:373)
	at io.quarkus.runner.recorded.VertxHttpProcessor$openSocket1873327713.deploy_0(Unknown Source)
	at io.quarkus.runner.recorded.VertxHttpProcessor$openSocket1873327713.deploy(Unknown Source)
	... 7 more
Caused by: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: Could not initialize class io.netty.util.internal.PlatformDependent
	at java.base@25.0.1/java.util.concurrent.CompletableFuture.wrapInExecutionException(CompletableFuture.java:345)
	at java.base@25.0.1/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:440)
	at java.base@25.0.1/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2094)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:901)
	... 10 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.netty.util.internal.PlatformDependent
	at io.netty.channel.group.DefaultChannelGroup.<init>(DefaultChannelGroup.java:48)
	at io.netty.channel.group.DefaultChannelGroup.<init>(DefaultChannelGroup.java:84)
	at io.vertx.core.net.impl.ServerChannelLoadBalancer.<init>(ServerChannelLoadBalancer.java:43)
	at io.vertx.core.net.impl.TCPServerBase.listen(TCPServerBase.java:230)
	at io.vertx.core.net.impl.TCPServerBase.bind(TCPServerBase.java:183)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:190)
	at io.vertx.core.http.HttpServer.listen(HttpServer.java:228)
	at io.vertx.core.http.HttpServer.listen(HttpServer.java:241)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.setupTcpHttpServer(VertxHttpRecorder.java:1351)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:1265)
	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$5(DeploymentManager.java:210)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:252)
	at io.vertx.core.impl.ContextInternal.lambda$runOnContext$0(ContextInternal.java:50)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@25.0.1/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.1/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:829)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:805)

$ echo $?
1
```

--- a/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyOnJava24ProcessorExtension.java
+++ b/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyOnJava24ProcessorExtension.java
@@ -2,7 +2,6 @@ package io.quarkus.netty.deployment;
 
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.ModuleOpenBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.NativeImageSystemPropertyBuildItem;
 
 /**
  * As Java 24 locks down access to sun.misc.Unsafe, Netty needs to adapt to this
@@ -17,11 +16,6 @@ import io.quarkus.deployment.builditem.nativeimage.NativeImageSystemPropertyBuil
  * setting such an option.
  */
 public class NettyOnJava24ProcessorExtension {
-
-    @BuildStep
-    public NativeImageSystemPropertyBuildItem jdk24CleanersViaSetAccessible() {
-        return new NativeImageSystemPropertyBuildItem("io.netty.tryReflectionSetAccessible", "true");
-    }
 
     @BuildStep
     ModuleOpenBuildItem openModules() {

